File: Submariner/SBTrackArtistNameTransformer.m
//
//  SBTrackArtistNameTransformer.m
//  Submariner
//
//  Created by Calvin Buckley on 2022-09-13.
//  Copyright © 2022 Calvin Buckley. All rights reserved.
//

#import "SBTrackArtistNameTransformer.h"

#import "Submariner-Swift.h"

@implementation SBTrackArtistNameTransformer


+ (Class)transformedValueClass {
    return [NSString class];
}

+ (BOOL)allowsReverseTransformation {
    return NO;
}

- (id)transformedValue:(id)value {
    SBTrack *track = (SBTrack*)value;
    // return album artist if we don't have an artist
    if (track.artistName == nil || [track.artistName isEqualToString: @""]) {
        return track.album.artist.itemName;
    }
    return track.artistName;
}

@end

File: Submariner/SBTrackArtistNameTransformer.swift
//
//  SBTrackArtistNameTransformer.swift
//  Submariner
//
//  Created by Calvin Buckley on 2023-04-23.
//  Copyright © 2023 Submariner Developers. All rights reserved.
//

import Cocoa

@objcMembers class SBTrackArtistNameTransformer: ValueTransformer {
    override class func allowsReverseTransformation() -> Bool {
        false
    }
    
    override class func transformedValueClass() -> AnyClass {
        return NSString.self
    }

    override func transformedValue(_ value: Any?) -> Any? {
        // value is NSNumber-based enum
        if let track = value as? SBTrack {
            if let artistName = track.artistName, artistName != "" {
                return artistName
            }
            return track.album?.artist?.itemName
        }
        return nil
    }
}
